lib/index.js

```diff
@@ -543,23 +545,65 @@ class Config {
       return
     }
 
+    const cliWorkspaces = this[_get]('workspaces', 'cli')
+
     for (const p of walkUp(this.cwd)) {
-      // walk up until we have a nm dir or a pj file
-      const hasAny = (await Promise.all([
-        stat(resolve(p, 'node_modules'))
-          .then(st => st.isDirectory())
-          .catch(() => false),
-        stat(resolve(p, 'package.json'))
-          .then(st => st.isFile())
-          .catch(() => false),
-      ])).some(is => is)
-      if (hasAny) {
+      const hasNodeModules = await stat(resolve(p, 'node_modules'))
+        .then((st) => st.isDirectory())
+        .catch(() => false)
+
+      const hasPackageJson = await stat(resolve(p, 'package.json'))
+        .then((st) => st.isFile())
+        .catch(() => false)
```

Comment:
what about handling root links? tbh I'm not sure what the supported usages are but I'm aware that arborist does support some for of root link node:

ref: https://github.com/npm/cli/blob/latest/workspaces/arborist/lib/arborist/build-ideal-tree.js#L384

Reply:
since we use `stat` here instead of `lstat` we'll resolve symlinks during the lookups and require that the `package.json` or `node_modules` correctly resolve to a file or directory, respectively. there's no change in behavior there so i think this should be just fine